### PR TITLE
fix(popover): adds a new `scale` prop

### PR DESCRIPTION
**Pull request by Elijbet**
**Related Issue:** #4628

## Summary

Adds scale to `popover` 

- so it's consistent with all other visual components that have a `scale` prop

- to coordinate with the scales of the `heading`, the optional dismissible nested `action` element, and the slotted x `icon`

- to ensure visual height consistency when the icon is collapsed

### CODE REVIEW

**Review by driskull — APPROVED** [review]
LGTM!

＼(＾O＾)／

File: src/components/popover/popover.scss [hunk 1]
```diff
@@ -1,6 +1,31 @@
 @include floatingUIHost();
 @include floatingUIArrow();
 
+// scale variants for child
```
Comment (jcfranco) [inline src/components/popover/popover.scss:4]: Let's drop this comment. The selectors are expressive enough.

File: src/components/popover/popover.stories.ts [hunk 2]
```diff
@@ -101,3 +101,29 @@ export const flipPlacements_TestOnly = (): string => html`
     document.querySelector(".my-popover").flipPlacements = ["right"];
   </script>
 `;
+
+export const scaleConsistencyPopoverHeadingActionSlottedIcon = stepStory(
```
Comment (jcfranco) [inline src/components/popover/popover.stories.ts:105]: Per https://github.com/Esri/calcite-components/blob/master/conventions/Documentation.md#screenshot-test-only-stories, please add the `_TestOnly` suffix.

File: src/components/popover/popover.stories.ts [hunk 3]
```diff
@@ -101,3 +101,29 @@ export const flipPlacements_TestOnly = (): string => html`
...
+  (): string => html`
+    <div style="width: 400px;">
+      ${referenceElementHTML}
+      <calcite-popover
+        heading="Dreams didn't make us kings. Dragons did. 🐉"
+        reference-element="reference-element"
+        placement="${select("placement", placements, defaultPopoverPlacement)}"
```
Comment (jcfranco) [inline src/components/popover/popover.stories.ts:112]: Please remove the knobs as these are not needed for test-specific stories.

File: src/components/popover/popover.stories.ts [hunk 4]
```diff
@@ -101,3 +101,29 @@ export const flipPlacements_TestOnly = (): string => html`
...
+        ${boolean("open", true)}
+        ${boolean("dismissible", true)}
+        scale="${select("scale", ["s", "m", "l"], "m")}"
+      >
+        ${contentHTML}
+      </calcite-popover>
+    </div>
+  `,
+  createSteps("calcite-popover")
```
Comment (jcfranco) [inline src/components/popover/popover.stories.ts:121]: Let's drop the stepped stories to keep our screenshot quota down.

**Review by jcfranco — APPROVED** [review]
Aside from a few comments, this LGTM! 🎉

File: src/components/popover/popover.stories.ts (hunk 4 above)
Comment (Elijbet) [reply src/components/popover/popover.stories.ts:121]: Ok, I added them for designers to be able to give me feedback on visuals. I'll prep a demo instead. I think having an interactive demo would be better for a design review anyway.

**Pull request merged**